### PR TITLE
Allow for multi SPIR-V shaders

### PR DESCRIPTION
**Pull request by dj2**
This CL updates AmberScript to allow SPIR-V to provide multiple shaders
in a single SHADER block. This is done with a 'multi' shader type and an
extension to ATTACH to specify the shader TYPE.

### CODE REVIEW

File: docs/amber_script.md [hunk 1]
```diff
@@ -31,11 +31,19 @@ which is the characters `0x` followed by hexadecimal digits.
...
 
 The compute pipeline can only contain compute shaders. The graphics pipeline
 can not contain compute shaders, and must contain a vertex shader and a fragment
 shader.
 
+The provided `multi` shader can only be used with `SPIRV-ASM` and `SPIRV-HEX`
+and allows for providing multiple shaders in a single module (so the `vertex`
+and `fragment` shaders can be provided together.)
+
+Note, `SPIRV-ASM` and `SPIRV-HEX` can also be used with each of the other shader
```
Comment (dneto0) [inline docs/amber_script.md]: I don't like this rule.  We often have many compute shaders in the same module which are then distinguished by entry point name.

File: docs/amber_script.md [hunk 2]
```diff
@@ -134,8 +142,13 @@ Shaders can be added into pipelines with the `ATTACH` call. Shaders may be
 attached to multiple pipelines at the same time.
 
 ```
+  # The provided shader for ATTACH must _not_ be a 'multi' shader.
```
Comment (dneto0) [inline docs/amber_script.md]: What happens with   compute shader from OpenCL-C?  Do we declare it as a 'compute' type but then we would also need to give the entry-point name.

File: docs/amber_script.md (hunk 1 above)
Comment (dj2) [reply docs/amber_script.md]: This rule doesn't preclude that? You can have a multi shader with a compute, vertex and fragment. You can't can't have a vertex and compute if it isn't multi. But, you can have a compute SPIRV-ASM as well.

File: docs/amber_script.md (hunk 2 above)
Comment (dj2) [reply docs/amber_script.md]: Ah, so you're saying we need to combine ENTRY_POINT and ATTACH, that makes sense.

File: docs/amber_script.md (hunk 1 above)
Comment (dneto0) [reply docs/amber_script.md]: Ok.  This works.  It's different from my mental model but it resolves to correct and unambiguous behaviour.

**Pull request merged**